File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
# 14, 16, 18, 20
def solution(bandage, health, attacks):
    answer = 0
    t, x, y = bandage[0], bandage[1], bandage[2] # 시전 시간, 초당, 추가
    now =  health # 현재 체력
    # 이전 공격 시간과 현재 공격 시간 사이 값
    # 0-2 : 1회복 = 데미지 / 2-9 : 6회복 / 9-10 : 0회복 
    last = 0
    for time, d in attacks:
        # 연속 성공 시간
        times = time - last - 1
        bonus = times // t
        # 현재 체력 계산 > max 넘는 경우 max로 변경
        now += times * x + bonus * y
        if now > health:
            now = health
        # 데미지 만큼 감소, 마지막 공격 타임
        now -= d
        last = time
        # 죽었는지 체크 => 죽었으면 break
        if now <= 0:
            now = -1
            break
    return now

File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
        int answer = 0;
        int t = bandage[0]; // 시전 시간
        int x = bandage[1]; // 초당 회복량
        int y = bandage[2]; // 추가 회복량
        int now = health; // 현재 체력
        int last = 0; // 초기값 0으로 설정
        // 공경을 순회
        for (int[] attack : attacks){
            int time = attack[0]; // 공격 시간
            int d = attack[1]; // 데미지
            int cont = time - last - 1; // 지속 시간
            now += x * cont + y * (cont / t); // 시간 만큼 초당 회복량 + 추가회복량
            now = (now > health) ? health : now; // max 넘으면 max로 맞춰줌
            now -= d; // 데미지만큼 감소시키기
            // 0보다 작으면 죽음, break
            if (now <= 0) {
                answer = -1;
                break;
            // answer를 now로 설정, 마지막 시간을 time으로
            } else {
                answer = now ;
                last = time;
            }
        }
        
        return answer;
    }
}